workspace/welcome-to-xod/112-standalone-internet/patch.xodp

```diff
@@ -0,0 +1,114 @@
+{
+  "comments": [
+    {
+      "content": "Here are these nodes:\n\n- `xod-dev/esp8266-mcu/connect`\n- `xod-dev/esp8266/connect`\n- `xod-dev/w5500/ethernet-shield`\n\n## Excercise\n\n1. Choose the proper node from the list above and replace the `tethering-inet` node with it.\n2. Bind the appropriate values in the node, if needed. It might be the name of a WiFi access point and password.\n3. Bind the proper `PORT` value to `servo`.\n4. Upload it without debug.\n\nNote that you need to login and generate an XOD Cloud API key for the project, if you did not (main menu: Edit -> Project Preferences).\n\nThe servo rotates to the angle that corresponds to the latest \"moisture\" value. Now you can open the chapter `015-xod-cloud` again, run a simulation and control your gauge from XOD IDE over the cloud.\n\nOr you can make another device, which will update this value automatically.\n\n---\n\nIn some cases, you might need to make an advanced configuration of the connected module or ethernet-shield. Check out examples in the library or read an article about [ethernet-shield advanced configuration](https://xod.io/docs/guide/w5500-advanced/).",
```

Comment:
Here are these nodes:

- `xod-dev/esp8266-mcu/connect`
- `xod-dev/esp8266/connect`
- `xod-dev/w5500/ethernet-shield`

## Excercise

1.  Replace the `tethering-inet` node with a suitable one from the list above.
2. Bind the appropriate values in the node, if it is necessary. It can be the name of a WiFi access point and password.
3. Bind the `PORT` value to `servo` node.
4. Upload the patch without debug mode.

Note that you need to login and generate an XOD Cloud API key for the project, only if you haven't done it before (main menu: Edit -> Project Preferences).

The servo rotates to the angle that corresponds to the latest "moisture" value. Now you can open the chapter `015-xod-cloud` again, run a simulation and control your gauge from XOD IDE over the cloud. Or you can make another device, which updates this value automatically.

---

In some cases, you might need to make an advanced configuration of the connected module or ethernet-shield. Check out examples in the library or read an article about [ethernet-shield advanced configuration](https://xod.io/docs/guide/w5500-advanced/).